bunsen-core/src/test/java/com/cerner/bunsen/mappings/ValueSetsTest.java

```diff
@@ -0,0 +1,498 @@
+
+    ValueSets valueSets = ValueSets.getEmpty(spark)
+        .withValueSets(valueSet(VALUE_SET_PREFIX + "a", "1", "a", "b", "c", "d"),
+            valueSet(VALUE_SET_PREFIX + "b", "1", "b", "c"),
+            valueSet(VALUE_SET_PREFIX + "c", "1", "c"),
+            valueSet(VALUE_SET_PREFIX + "d", "1", "d"),
+            valueSet(VALUE_SET_PREFIX + "e", "1", "e", "f"),
+            valueSet(VALUE_SET_PREFIX + "f", "1", "f"));
+
+    List<Value> values = valueSets.getValues().collectAsList();
+
+    ValueSet isA = valueSets.getValueSet(VALUE_SET_PREFIX + "a", "1");
+    ValueSet isB = valueSets.getValueSet(VALUE_SET_PREFIX + "b", "1");
+    ValueSet isE = valueSets.getValueSet(VALUE_SET_PREFIX + "e", "1");
+
+    Assert.assertEquals(VALUE_SET_CLOSURE.size(), values.size());
+    Assert.assertTrue(VALUE_SET_CLOSURE.containsAll(values));
+
+    Assert.assertEquals(4, isA.getCompose().getIncludeFirstRep().getConcept().size());
+    Assert.assertEquals(2, isB.getCompose().getIncludeFirstRep().getConcept().size());
+    Assert.assertEquals(2, isE.getCompose().getIncludeFirstRep().getConcept().size());
+
+    Assert.assertEquals(6, valueSets.getValueSets().count());
+  }
+
+  @Test
+  public void testFromDirectory() {
+    ValueSets valueSets = ValueSets.getEmpty(spark)
+        .withValueSetsFromDirectory("src/test/resources/valuesets");
+
+    ValueSet genderValueSet = valueSets.getValueSet("urn:cerner:bunsen:valueset:gender", "0.0.1");
+
+    Assert.assertNotNull(genderValueSet);
+    Assert.assertEquals("urn:cerner:bunsen:valueset:gender", genderValueSet.getUrl());
+    Assert.assertEquals("0.0.1", genderValueSet.getVersion());
+  }
+
+  @Test
+  public void testWriteToNewTables() {
+
+    String database = "test_valuesets_write";
+    spark.sql("CREATE DATABASE " + database);
+
+    ValueSets valueSets = ValueSets.getEmpty(spark)
+        .withValueSetsFromDirectory("src/test/resources/valuesets");
+
+    valueSets.writeToDatabase(database);
+
+    ValueSets reloadedValueSets = ValueSets.getFromDatabase(spark, database);
+
+    ValueSet genderValueSet = reloadedValueSets
+        .getValueSet("urn:cerner:bunsen:valueset:gender", "0.0.1");
+
+    Assert.assertNotNull(genderValueSet);
+    Assert.assertEquals("urn:cerner:bunsen:valueset:gender", genderValueSet.getUrl());
+    Assert.assertEquals("0.0.1", genderValueSet.getVersion());
+    Assert.assertEquals(3, genderValueSet.getCompose().getInclude().size());
+  }
+
+  @Test
+  public void testValueSetsIncludesNoConcepts() {
+
+    ValueSets valueSets = ValueSets.getEmpty(spark)
+        .withValueSets(valueSet("urn:cerner:valueset:valueset1", "1"),
+            valueSet("urn:cerner:valueset:valueset2", "1"));
+
+    valueSets.getValueSets()
+        .collectAsList()
+        .forEach(valueSet -> valueSet.getCompose()
+            .getInclude()
+            .forEach(inclusion -> Assert.assertTrue(inclusion.getConcept().isEmpty())));
+  }
+
+  @Test
+  public void testValuesAreSorted() {
```

Comment:
Here I was wanting to enforce the same sorting behavior that I thought was described in [`ConceptMapsTests`](https://github.com/cerner/bunsen/blob/master/bunsen-core/src/test/java/com/cerner/bunsen/mappings/ConceptMapsTest.java#L151), but that I noticed the test didn't fully describe sorting behavior since it lacks an additional group to sort against. It would actually appear that since groups are [object-copied](https://github.com/cerner/bunsen/blob/master/bunsen-core/src/main/java/com/cerner/bunsen/mappings/ConceptMaps.java#L646) when removing the concepts, whatever order groups were specified is kept, even if that order isn't lexicographically ascending. 

If keeping a sorting contract isn't necessary, we can remove the tests that would check it from both `ConceptMapsTests` and here.